dataplane/reconcilers/routes.go

```diff
@@ -0,0 +1,255 @@
+			})
+			hopID = group.Oid
+			if err != nil {
+				log.Warningf("failed to create next hop group: %v", err)
+				return ygnmi.Continue
+			}
+			for i, nh := range route.GetNextHops().GetHops() {
+				hopID, err = ni.createNextHop(ctx, nh)
+				if err != nil {
+					log.Warningf("failed to create next hop: %v", err)
+					return ygnmi.Continue
+				}
+				_, err = ni.nextHopGroupClient.CreateNextHopGroupMember(ctx, &saipb.CreateNextHopGroupMemberRequest{
+					Switch:         ni.switchID,
+					NextHopGroupId: &group.Oid,
+					NextHopId:      &hopID,
+					Weight:         proto.Uint32(uint32(route.GetNextHops().Weights[i])),
+				})
+				if err != nil {
+					log.Warningf("failed to create next group member: %v", err)
+					return ygnmi.Continue
+				}
+			}
+		}
+		rReq.NextHopId = proto.Uint64(hopID)
+		if _, err := ni.routeClient.CreateRouteEntry(ctx, &rReq); err != nil {
+			log.Warningf("failed to create route: %v", err)
+			return ygnmi.Continue
+		}
+		log.Infof("created route entry: %v", &rReq)
+
+		return ygnmi.Continue
+	})
+	go func() {
+		// TODO: handle error
+		if _, err := w.Await(); err != nil {
+			log.Warningf("routes watch err: %v", err)
+		}
+	}()
+	ni.closers = append(ni.closers, cancelFn)
+	return nil
+}
+
+func (ni *Reconciler) createNextHop(ctx context.Context, hop *dpb.NextHop) (uint64, error) {
+	ip, err := netip.ParseAddr(hop.GetNextHopIp())
+	if err != nil {
+		return 0, err
+	}
+	data := ni.ocInterfaceData[ocInterface{name: hop.GetInterface().GetInterface(), subintf: hop.GetInterface().GetSubinterface()}]
+	hopReq := saipb.CreateNextHopRequest{
+		Switch:            ni.switchID,
+		Type:              saipb.NextHopType_NEXT_HOP_TYPE_IP.Enum(),
+		Ip:                ip.AsSlice(),
+		RouterInterfaceId: proto.Uint64(data.rifID),
+	}
+	resp, err := ni.nextHopClient.CreateNextHop(ctx, &hopReq)
+	if err != nil {
+		return 0, err
+	}
+	if hop.GetGue() != nil {
+		acts, err := gueActions(hop.GetGue())
+		if err != nil {
+			return 0, err
+		}
+		// TODO: Ideally, this would use the SAI tunnel, but it's not currently supported.
+		_, err = ni.fwdClient.TableEntryAdd(ctx, &fwdpb.TableEntryAddRequest{
+			ContextId: &fwdpb.ContextId{Id: ni.contextID},
+			TableId:   &fwdpb.TableId{ObjectId: &fwdpb.ObjectId{Id: engine.NHActionTable}},
+			EntryDesc: &fwdpb.EntryDesc{Entry: &fwdpb.EntryDesc_Exact{
+				Exact: &fwdpb.ExactEntryDesc{
+					Fields: []*fwdpb.PacketFieldBytes{{
+						FieldId: &fwdpb.PacketFieldId{Field: &fwdpb.PacketField{FieldNum: fwdpb.PacketFieldNum_PACKET_FIELD_NUM_NEXT_HOP_ID}},
+						Bytes:   binary.BigEndian.AppendUint64(nil, resp.Oid),
+					}},
+				},
+			}},
+			Actions: acts,
+		})
+		if err != nil {
+			return 0, err
+		}
+	}
+	return resp.Oid, nil
+}
+
+func gueActions(gueHeaders *dpb.GUE) ([]*fwdpb.ActionDesc, error) {
+	var ip gopacket.SerializableLayer
+	var headerID fwdpb.PacketHeaderId
+	if len(gueHeaders.SrcIp) == 4 {
```

Comment:
I think this assumption might be wrong given you're just copying the bytes in sysrib/server.go
https://go.dev/play/p/UaIO9bPMhAT

Locally I have BGP-triggered GUE test failing. Not sure why prefix_set test also failed but perhaps a non-deterministic failure.